independent-projects/resteasy-reactive/common/processor/src/main/java/org/jboss/resteasy/reactive/common/processor/EndpointIndexer.java

```diff
@@ -356,6 +358,86 @@ protected abstract METHOD createResourceMethod(MethodInfo info, ClassInfo actual
+                existingClassNameBindings, considerApplication, httpAnnotationToMethod, index, applicationScanningResult,
+                getAnnotationStore());
+        Map<String, BasicResourceClassInfo> basicResourceClassInfoMap = new HashMap<>();
+        for (FoundEndpoint endpoint : endpoints) {
+
+            // assemble method path
+            String methodPath = readStringValue(getAnnotationStore().getAnnotation(endpoint.methodInfo, PATH));
+            if (endpoint.httpMethod != null) {
+                // HTTP annotation method
+                validateHttpAnnotations(endpoint.methodInfo);
+                if (methodPath != null) {
+                    if (!methodPath.startsWith("/")) {
+                        methodPath = "/" + methodPath;
+                    }
+                    if (methodPath.endsWith("/")) {
+                        methodPath = handleTrailingSlash(methodPath);
+                    }
+                } else {
+                    methodPath = "";
+                }
+            } else {
+                // resource locator method
+                if (methodPath != null) {
+                    if (!methodPath.startsWith("/")) {
+                        methodPath = "/" + methodPath;
+                    }
+                    if (methodPath.endsWith("/")) {
+                        methodPath = methodPath.substring(0, methodPath.length() - 1);
+                    }
+                }
+            }
+
+            // create BasicResourceClassInfo and remember it
+            BasicResourceClassInfo basicResourceClassInfo = basicResourceClassInfoMap.get(endpoint.classInfo.name().toString());
+            if (basicResourceClassInfo == null) {
+                String[] classProduces = extractProducesConsumesValues(
+                        getAnnotationStore().getAnnotation(endpoint.classInfo, PRODUCES));
+                String[] classConsumes = extractProducesConsumesValues(
+                        getAnnotationStore().getAnnotation(endpoint.classInfo, CONSUMES));
+                basicResourceClassInfo = new BasicResourceClassInfo(resourceClassPath, classProduces,
+                        classConsumes, pathParameters, getStreamAnnotationValue(endpoint.classInfo));
+                basicResourceClassInfoMap.put(endpoint.classInfo.name().toString(), basicResourceClassInfo);
+            }
+
+            ResourceMethod method = createResourceMethod(endpoint.classInfo, actualEndpointInfo, basicResourceClassInfo,
+                    endpoint.classNameBindings, endpoint.httpMethod, endpoint.methodInfo, methodPath);
+            ret.add(method);
+        }
+
+        return ret;
+    }
+
+    /**
+     * Return endpoints defined directly on classInfo.
+     *
+     * @param classInfo resource class
+     * @return classInfo endpoint descriptors
+     */
+    public static Collection<MethodDescriptor> collectClassEndpoints(ClassInfo classInfo,
```

Comment:
Why is `MethodDescriptor` being used here? It's weird to use a Gizmo class during indexing...

Reply:
It has nicely implemented `equals`, `MethodInfo` has more complex `equals` and it's bad idea to rely on `equals` implementation as it can change. I can replace `MethodDescriptor` with string that will join fully qualified class name and method signature with a white space. Do you have different suggestion? Later it's important to have a way to determine if "this method is the one we want secure".

Reply:
Ah, I think I know how to fix this. I'll do it now.

Reply:
It turned out that I didn't need it here at all, thanks.